### PR TITLE
create payment model

### PR DESCRIPTION
**Pull request by LeeRenJie**
depends on #261

### CODE REVIEW

File: photo-app/app/models/payment.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Payment < ApplicationRecord
+  attr_accessor :card_number, :card_cvv, :card_expires_month, :card_expires_year
+  belongs_to :user
```
Comment (tlkiong) [inline photo-app/app/models/payment.rb:3]: Please organize your code properly. Have an extra space for each section. don't just jumble up together. That makes it hard to read

File: photo-app/app/models/payment.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  def self.month_options
```
Comment (tlkiong) [inline photo-app/app/models/payment.rb:4]: What do you mean by `month_options`?

File: photo-app/app/models/payment.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  def self.year_options
+    (Date.today.year..(Date.today.year+10)).to_a
+  end
+
+  def process_payment
+    customer = Stripe::Customer.create email: email, card: token
+    Stripe::Charge.create customer: customer.id,
+    amount: 1000,
+    description: 'Premium',
+    currency: 'usd'
```
Comment (tlkiong) [inline photo-app/app/models/payment.rb:17]: Look at this properly Does it clearly illustrate where is the start & end of the argument you passing in? Where is the method? Think please. THink of others. Not just yourself. Don't be so selfish

File: photo-app/app/models/user.rb [hunk 4]
```diff
@@ -3,4 +3,6 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable, :confirmable,
          :recoverable, :rememberable, :validatable, :trackable
+  has_one :payment
```
Comment (tlkiong) [inline photo-app/app/models/user.rb:6]: huh? 1 payment record only?
The user cannot make many payments?

File: photo-app/db/migrate/20220122130633_create_payments.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class CreatePayments < ActiveRecord::Migration[6.1]
+  def change
+    create_table :payments do |t|
+      t.string :email
+      t.string :token
```
Comment (tlkiong) [inline photo-app/db/migrate/20220122130633_create_payments.rb:5]: use text.....

**Review by tlkiong — CHANGES_REQUESTED** [review]

File: photo-app/app/models/payment.rb (hunk 2 above)
Comment (LeeRenJie) [reply photo-app/app/models/payment.rb:4]: `self.month_options`, `self.year_options` are used in the sign-up form for the users to select the month and year. The method `self.month_options` returns following to use in the form

```
[["1 - January", 1],
 ["2 - February", 2],
 ["3 - March", 3],
 ["4 - April", 4],
 ["5 - May", 5],
 ["6 - June", 6],
 ["7 - July", 7],
 ["8 - August", 8],
 ["9 - September", 9],
 ["10 - October", 10],
 ["11 - November", 11],
 ["12 - December", 12]]
```

`self.year_options` returns:
`[2022, 2023, 2024, 2025, 2026, 2027, 2028] `

File: photo-app/app/models/user.rb (hunk 4 above)
Comment (LeeRenJie) [reply photo-app/app/models/user.rb:6]: Ya cuz they only need to make the payment once when they sign up to the site

File: photo-app/app/models/payment.rb (hunk 2 above)
Comment (tlkiong) [reply photo-app/app/models/payment.rb:5]: My point with this question is that the naming is not clear

File: photo-app/app/models/user.rb (hunk 4 above)
Comment (tlkiong) [reply photo-app/app/models/user.rb:6]: what if later on if the business side decide that they want to change it into a subscription?
Or, based on the current logic, what if a payment fail?

File: photo-app/app/models/payment.rb (hunk 2 above)
Comment (tlkiong) [reply photo-app/app/models/payment.rb:5]: Not planning to change the naming?

**Review by tlkiong — APPROVED** [review]

**Pull request merged**